### PR TITLE
(feat) TypeScript plugin

### PR DESCRIPTION
**Pull request by dummdidumm**
Initially support
- rename (doesn't work for prop renames yet, I'm okay with that)
- diagnostics
- find references
- go to definition

This makes all files TSX hardcoded for now, it seems the TS server is okay with importing tsx into js.

To test this, add 
```
        "typescriptServerPlugins": [
            {
                "name": "typescript-svelte-plugin",
                "enableForWorkspaceTypeScriptVersions": true,
                "path": "typescript-plugin"
            }
        ],
```

to the `contributes` property in the vscode extension's `package.json`

In contrast to #842 this does not set `"languages": ["svelte"]` which means TS will not take over Svelte files. The advantage is that diagnostics, rename etc within Svelte files stays in the control of the language-server, and only inside TS/JS files Svelte support is added. The drawback is that you need to save file changes to disk for Svelte files in order for the TS server to notice it.

Next steps after merge:
- publish plugin to npm
- make it opt-in in the VS Code extension
- check if we can move some common stuff into a shared folder in the monorepo; maybe we need to publish another npm package

#580
#550
#342
#110

@jasonlyu123 would be great if you could take a look 😃

### CODE REVIEW

File: packages/typescript-plugin/src/language-service/definition.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+export function decorateGetDefinition(
+    ls: ts.LanguageService,
+    snapshotManager: SvelteSnapshotManager,
+    logger: Logger
+): void {
+    const getDefinitionAndBoundSpan = ls.getDefinitionAndBoundSpan;
+    ls.getDefinitionAndBoundSpan = (fileName, position) => {
+        const definition = getDefinitionAndBoundSpan(fileName, position);
+        if (!definition || !definition.definitions) {
```
Comment (jasonlyu123) [inline packages/typescript-plugin/src/language-service/definition.ts:14]: Maybe `if (!definition?.definitions`)? You can ignore this if you think the original code is more clear.

**Review by jasonlyu123 — CHANGES_REQUESTED** [review]
Looks good. It's indeed a joy to see the svelte files show up in the ts language features. I listed some minor issues. Another feature I think can be decorated is `getImplementationAtPosition`. It's mostly the same as `getDefinition`. Should be easy to add.

**Comment by dummdidumm** [comment]
When does `getImplementationAtPosition` get called? How could I test this?

**Comment by jasonlyu123** [comment]
https://code.visualstudio.com/docs/editor/editingevolved#_go-to-implementation
This one. Or maybe we can do it in another PR.

**Pull request merged**